### PR TITLE
feat: add simple voting functionality

### PR DESCRIPTION
**Pull request by kyron321**

### CODE REVIEW

File: src/components/Votes.jsx
Comment (B7Bprog) [inline src/components/Votes.jsx:19]: Need to use optimistic rendering here.

File: src/components/Votes.jsx
Comment (B7Bprog) [inline src/components/Votes.jsx]: Please remove all logs.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Kyron, please implement optimistic rendering, so the user can see the updated vote immediately, even before the patch request finishes.

Also it's nice that a user can only vote once, but would it not be more appropriate if they could only vote once in one direction and still can revoke it? So when they upvote and change their mind, then they could downvote too, but could only alter the original votes by 1. I leave this up to you to think about and decide. 

Please implement optimistic rendering and send us a quick PR when it's done.

**Review by roisinoreilly — APPROVED** [review]
- I love the UI, colour scheme and icons, it all looks very sleek
- nice mobile design, the view comments button just needs altering slightly in this view or the text gets pushed outside it
- when you downvote after upvoting it doesn't remove the original vote/let you downvote twice, you should be able to vote up, vote down to return the votes to what they originally were and undo your vote, then vote down again
- it's good that you alert the user that there is a network error if their vote is unsuccessful, but it might be a good idea to still render the votes button to let them try again without refreshing

Really great work keep it up, try and get the votes working as I've suggested, after that merge away

**Pull request merged**
